site/docs/patterns/forms.mdx

```diff
@@ -0,0 +1,839 @@
+    <ul style="margin-bottom:0;">
+        <li><strong>If the form has a main heading, it <em>must</em> be the first element</strong>
+            <ul style="margin-top:0;">
+                <li>The main heading <em>should</em> preferably be h1 element, even if visually styled as a smaller level heading.</li>
+            </ul>
+        </li>
+        <li>The the sub-heading <em>should</em> placed next after the main heading, and preferably be a h2 element.</li>
+        <li><strong>Keep it short and descriptive!</strong>
+            <ul style="margin-top:0;">
+                <li>Main heading <em>should</em> be less than five words.</li>
+                <li>Sub-heading <em>should</em> be less than two to three sentences.</li>
+            </ul>
+        </li>
+    </ul>
+</div>
+
+---
+
+### 2. Form wide instructions
+To help the user better understand the functionality of the form, instructions and other **information about filling the form and required information** should be provided in the beginning of the form. 
+
+<Image src="../../static/patterns/form/form-guiding-instructions@2x.png" alt="Form wide instructions" style="max-width:432px;" viewable /> 
+
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <div class="guideline-do-label" style="color:var(--color-success); margin-bottom:var(--spacing-s);"><IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle'}} size="s"/><strong>Do</strong></div>
+    <ul style="margin-bottom:0;">
+        <li>The instructions should be placed after the heading and sub-heading, before the first form control or fieldeset.</li>
+        <li>The inline notification component should be used for form wide instructions.</li>
+    </ul>
+</div> 
+
+---
+
+### 3. Contextual instructions
+**Instructions related to specific form component** can be provided for the user in three different ways. They each have different uses and levels of importance:
+- **Assistive text** provides essential information 
+- **Tooltip** provides context or background information
+- **Placeholder** acts as a supplementary visual aid
+
+#### a) Assistive text
+A short secondary description that elaborates the label and helps the user to provide the **right information in the right format**.
+
+**The assistive text is always visible to the user, so any *essential* information the user needs to complete their task *must* be provided as assistive text.**
+- Displayed below the form control
+- Screen readers read the assistive text after the label when focusing the form element.
+
+<Image src="../../static/patterns/form/form-guiding-assistive@2x.png" alt="Assistive text" style="max-width:376px;" viewable /> 
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
```

Comment:
Onko meillä jotain ohjetta siihen, että mitä assistiveen ei pitäisi laittaa? Periaatteessa joku voi pohtia, että onko se sitten paikka nice-to-have -tiedolle, jos must-have tietoa ei tarvitse esittää? Vai pitäisikö tuollainen tieto aina laittaa esim. tooltipin alle?

Reply:
Hyvä kysymys. Ehkä tätä ei kannata lähteä määrittelemän liian tarkasti, kun linjanvedon pakollisen ja nice-to-have -tiedon välillä voi olla aika häilyvä ja kontekstista riippuvainen. 

Ehkä tähän voisi lisätä ohjeeksi jotain tyyliin "ylimääräinen tieto voi aiheuttaa kognitiivista overloadia, eli älä nyt herran jestas tunge ihan joka kentän yhteyteen jotain liibalaabaa". 😂

Reply:
Lisäsin: "The assistive text should only be used for essential information. Avoid causing unnecessary cognitive overload for the user by giving too much information."